acvm-repo/bn254_blackbox_solver/src/pedersen/commitment.rs

```diff
@@ -0,0 +1,64 @@
+// Taken from: https://github.com/laudiacay/barustenberg/blob/df6bc6f095fe7f288bf6a12e7317fd8eb33d68ae/barustenberg/src/crypto/pedersen/pederson.rs
+
+use ark_ec::{short_weierstrass::Affine, AffineRepr, CurveGroup};
+use ark_ff::PrimeField;
+use grumpkin::{Fq, Fr, GrumpkinParameters};
+
+use crate::generator::generators::{derive_generators, DEFAULT_DOMAIN_SEPARATOR};
+
+/// Given a vector of fields, generate a pedersen commitment using the indexed generators.
+pub(crate) fn commit_native_with_index(
+    inputs: &[Fq],
+    starting_index: u32,
+) -> Affine<GrumpkinParameters> {
+    let generators =
+        derive_generators(DEFAULT_DOMAIN_SEPARATOR, inputs.len() as u32, starting_index);
+
+    inputs.iter().enumerate().fold(Affine::zero(), |mut acc, (i, input)| {
+        //TODO: this is a sketch conversion do better
+        acc = (acc + (generators[i] * Fr::from_bigint(input.into_bigint()).unwrap()).into_affine())
+            .into_affine();
+        acc
+    })
+}
+
+#[cfg(test)]
+mod test {
+
+    use ark_ec::short_weierstrass::Affine;
+    use ark_ff::MontFp;
+    use ark_std::{One, Zero};
+    use grumpkin::Fq;
+
+    use crate::pedersen::commitment::commit_native_with_index;
+
+    #[test]
+    fn commitment() {
+        let res = commit_native_with_index(&[Fq::one(), Fq::one()], 0);
+        let expected = Affine::new(
+            // 2f7a8f9a6c96926682205fb73ee43215bf13523c19d7afe36f12760266cdfe15
+            MontFp!(
+                "21475250338311530111088781112432132511855209292730670949974692984887182229013"
```

Comment:
Any reason we decided to use the mont form rather than hex here?

Reply:
Mostly that it's annoying to create an `Fq` from hex without going through `FieldElement` which I wanted to avoid. I've just used `FieldElement` however.